app/controllers/auth/MergePageAuthAction.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.auth
+
+import com.google.inject.{ImplementedBy, Inject}
+import config.ApplicationConfig
+import play.api.mvc.Results.Redirect
+import play.api.mvc._
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.retrieve.~
+import uk.gov.hmrc.auth.core.{Nino => AuthNino, _}
+import uk.gov.hmrc.domain.{Nino, SaUtr, Uar}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+import uk.gov.hmrc.play.bootstrap.auth.DefaultAuthConnector
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MergePageAuthActionImpl @Inject()(
+  override val authConnector: DefaultAuthConnector,
+  cc: MessagesControllerComponents)(implicit ec: ExecutionContext, appConfig: ApplicationConfig)
+    extends MergePageAuthAction with AuthorisedFunctions {
+
+  override val parser: BodyParser[AnyContent] = cc.parsers.defaultBodyParser
+  override protected val executionContext: ExecutionContext = ec
+
+  override def invokeBlock[A](request: Request[A], block: AuthenticatedRequest[A] => Future[Result]): Future[Result] = {
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    authorised(ConfidenceLevel.L50 or (Enrolment("IR-SA") or Enrolment("IR-SA-AGENT")) or AuthNino(hasNino = true))
```

Comment:
You can remove all of these: 
`(Enrolment("IR-SA") or Enrolment("IR-SA-AGENT")) or AuthNino(hasNino = true)`

Reply:
removed